src/exe/Main.hs

```diff
@@ -88,7 +89,11 @@ application = do
       inputFiles <- getOpt optInputFiles
       if showHelp || null inputFiles
         then embed putUsageInfo
-        else runInputFile $ mapM_ processInputFile inputFiles
+        else 
+          -- TODO There probably is a better way to add actual input files to the
+          -- Report effect
+          runInputFile inputFiles . reportToHandleOrCancel stderr
+          $ mapM_ processInputFile inputFiles
```

Comment:
That's the reason I didn't change the `InputFile` effect to use [Polysemy's `Reader` effect](https://hackage.haskell.org/package/polysemy-1.3.0.0/docs/Polysemy-Reader.html) initially, but used the `State` effect with an `addInputFile` method.
Can this somehow be implemented with the `Reader` effect (possibly with the unexplained `Local` action)? I thought (not) being able to change the state of the effect was the main difference between the `State` and `Reader` effect.
I can of course change `InputFile` to use the `State` effect once again.

Reply:
You can use Polysemy's `State` effect internally instead of the `Reader` effect even if the `InputFile` effect itself appears like a reader effect to the user since no equivalent to the `put` operation is exposed.